doc/Stack_and_VS_Code.md

```diff
@@ -0,0 +1,115 @@
+for users of the Cabal tool.
+
+If the VS Code extension is set not to use GHCup, its user needs to ensure that
+each version of HLS that the extension needs is on the PATH.
+
+For the most part, the versions of HLS provided by GHCup are built with the same
+versions of GHC that Stack downloads from its default `setup-info` dictionary
+(see [YAML configuration: setup-info](yaml_configuration.md)). Stack's default
+is to mirror the 'official' binary distributions published by GHC. However, in
+some cases (hopefully rare), it is possible that a GHCup-supplied HLS has been
+built with a different binary distribution of GHC than the one which Stack has
+installed.
+
+One example of that occurred with the release of GHC 9.0.2. For some Linux users
+(Debian 9 and Fedora 27), the version of GHC 9.0.2 linked on GHC’s download web
+page was broken. However, elsewhere, alternative ‘9.0.2a’. versions became
+available. For a while, Stack referred to the versions published by GHC while
+the GHCup-supplied versions of HLS were built using alternative versions. This
+incompatibility lead to problems. It was resolved by Stack's default also being
+changed to refer to the '9.0.2a' versions. (Where Stack has already installed
+GHC 9.0.2, it is necessary to delete GHC 9.0.2 from the `stack path --programs`
+directory. This will cause Stack to reinstall the alternative version, when it
+first needs GHC 9.0.2. Stack should distinguish what it builds with the
+alternative from what it has build, and cached, with the original GHC 9.0.2.)
+
+### Workaround #1
+
+One workaround is to allow GHCup to install versions of GHC on the PATH and to
+cause Stack to use those versions of GHC, by making use of Stack's `install-ghc`
+option (which needs to be disabled) and Stack's `system-ghc` option (which needs
+to be enabled) (see [YAML configuration](yaml_configuration.md)).
+
+For this workaround to work, each time that a resolver is used that references a
+different version of GHC, GHCup must be used to switch to that version of GHC.
+For example, if using `resolver: lts-19.16` (GHC 9.0.2), the command
+`ghcup set ghc 9.0.2` must be used to set GHC 9.0.2. That may be inconvenient
+and undermines one the primary benefits of Stack over other Haskell build tools,
+namely that Stack automatically ensures that the necessary version of GHC is
+available.
+
+It is hoped that the next release of Stack will include a new feature that will
+allow the VS Code extension/GHCup to customise Stack, so that Stack can
+automatically switch between versions of GHC that GHCup has downloaded and
+installed. It is hoped a future release of the extension/GHCup will take
+advantage of that new feature. See
+[Stack #5585](https://github.com/commercialhaskell/stack/pull/5585) and
+[GHCup #392](https://gitlab.haskell.org/haskell/ghcup-hs/-/issues/392).
+
+### Workaround #2
+
+Another workaround is to install GHCup so that is 'empty' except for the current
```

Comment:
This is not a worakaround. It will cause the ABI mismatch problems, whenever stack supplies a different bindist that HLS was built with.